### DISCUSSION TITLE
Not able to create tables in my new Supabase Project

### DISCUSSION THREAD

**Discussion by 345ishaan** [supabase/supabase#31129] [Questions]
Hi,
I am trying the test example under a new project i created but getting the following error:

```
    raise APIError(r.json())
postgrest.exceptions.APIError: {}
```
Here is the code:
```
supa_client = create_client(url, key)

record = {
    "id": 1,  # Ensure this matches the expected data type
    "name": "Denmark"
}

try:
    response = supa_client.table('countries').insert([record]).execute()
    if response.error:
        print("Error inserting data:", response.error)
    else:
        print("Data inserted successfully:", response.data)
except Exception as e:
    print("Exception occurred:", e)
    ```

**Comment by GaryAustin1** [comment]
Not sure what language you are using, but what comes back in response.error?  There should be more info than just API error.
Check your Supabase dashboard API Gateway log and see if the request is making it there.

**Reply by 345ishaan** [reply]
I am using python API. It seems to be failing in `execute`, i guess inability to parse the request.

```
import os
import json
import supabase
print(supabase.__version__)
from supabase import create_client, Client

url: str = os.environ.get("SUPABASE_URL")
key: str = os.environ.get("SUPABASE_KEY")

print(url, key)

supa_client = create_client(url, key)

record = {
    "id": 1,  # Ensure this matches the expected data type
    "name": "Denmark"
}

response = supa_client.table('countries').insert([record]).execute()
print(response.error)
```

```Traceback (most recent call last):
  File "/Users/....", line 20, in <module>
    response = supa_client.table('countries').insert([record]).execute()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/...", line 78, in execute
    raise APIError(r.json())
postgrest.exceptions.APIError: {}```

**Reply by GaryAustin1** [reply]
You did create a countries table with the columns id and name?

**Reply by 345ishaan** [reply]
aah no i did not create any table. do i need to do that manually? I was expecting to create if it does not exist.

**Reply by GaryAustin1** [reply]
You have to create all tables (the docs under the data tab show you what you need for the sample code).
You will also have to enable/disable RLS and eventually add RLS policies to limit what user's do.

**Reply by 345ishaan** [reply]
Thanks, not related. but is there any easy way to delete all rows? I tried to use `._in` to do batch deletion but i can't do it in one shot.

**Reply by GaryAustin1** [reply]
From the UI use the SQL editor.  From the API you need to come up with a filter on a row that will always be met. Usually using the .not or .neq operator on a value you know can't be there.  For safety it is setup to require a filter to keep you from accidentally deleting all rows.
